tests/GithubApp.py

```diff
@@ -99,8 +102,17 @@ def testGetPublicApp(self):
         self.assertEqual(app.url, "/apps/github-actions")
 
     def testGetAuthenticatedApp(self):
-        # For this to work correctly in record mode, this test must be run with --auth_with_jwt
-        app = self.g.get_app()
+        auth = github.Auth.AppAuth(APP_ID, PRIVATE_KEY)
+        g = github.Github(auth=auth)
+
+        with self.assertWarns(DeprecationWarning) as warning:
+            app = g.get_app()
+
+            self.assertWarningIn(
```

Comment:
```suggestion
            self.assertWarning(
```

Reply:
The problem is that httpretty/core generates another warning, see https://github.com/PyGithub/PyGithub/pull/2549#issuecomment-1587869573

Reply:
Which Python version are you using?